apps/prairielearn/assets/scripts/instructorAssessmentManualGradingAssessmentQuestionClient.ts

```diff
@@ -0,0 +1,404 @@
+              })}"
+            >
+              <i class="fas fa-user-tag"></i>
+              Assign to: ${grader.name || ''} (${grader.uid})
+            </button>
+          `,
+        )}
+        <button
+          class="dropdown-item"
+          type="submit"
+          name="batch_action_data"
+          value="${JSON.stringify({ assigned_grader: null })}"
+        >
+          <i class="fas fa-user-slash"></i>
+          Remove grader assignment
+        </button>
+        <div class="dropdown-divider"></div>
+        <button
+          class="dropdown-item"
+          type="submit"
+          name="batch_action_data"
+          value="${JSON.stringify({ requires_manual_grading: true })}"
+        >
+          <i class="fas fa-tag"></i>
+          Tag as required grading
+        </button>
+        <button
+          class="dropdown-item"
+          type="submit"
+          name="batch_action_data"
+          value="${JSON.stringify({ requires_manual_grading: false })}"
+        >
+          <i class="fas fa-check-square"></i>
+          Tag as graded
+        </button>
+      </div>
+    </div>
+  `.toString();
+}
+
+function updateGradingTagButton() {
+  $('.grading-tag-button').prop(
+    'disabled',
+    !$('#grading-table').bootstrapTable('getSelections').length,
+  );
+}
+
+function pointsFormatter(
+  row: InstanceQuestionRow,
+  field: 'manual_points' | 'auto_points' | 'points',
+  hasCourseInstancePermissionEdit: boolean,
+  urlPrefix: string,
+  csrfToken: string,
+) {
+  const points = row[field];
+  const maxPoints = row.assessment_question[`max_${field}`];
+  const buttonId = `editQuestionPoints_${field}_${row.id}`;
+
+  return html`${formatPoints(points ?? 0)}
+    <small>/<span class="text-muted">${maxPoints ?? 0}</span></small>
+    ${hasCourseInstancePermissionEdit
+      ? EditQuestionPointsScoreButton({
+          field,
+          instance_question: row,
+          assessment_question: row.assessment_question,
+          urlPrefix: urlPrefix ?? '',
+          csrfToken: csrfToken ?? '',
+          buttonId,
+        })
+      : ''}`;
+}
+
+function scorebarFormatter(
+  score: number | null,
+  row: InstanceQuestionRow,
+  hasCourseInstancePermissionEdit: boolean,
+  urlPrefix: string,
+  csrfToken: string,
+) {
+  const buttonId = `editQuestionScorePerc${row.id}`;
+
+  return html`<div class="d-inline-block align-middle">
+      ${score == null
+        ? ''
+        : html`<div class="progress bg" style="min-width: 10em; max-width: 20em;">
+            <div
+              class="progress-bar bg-success"
+              style="width: ${Math.floor(Math.min(100, score))}%"
+            >
+              ${score >= 50 ? `${Math.floor(score)}%` : ''}
+            </div>
+            <div
+              class="progress-bar bg-danger"
+              style="width: ${100 - Math.floor(Math.min(100, score))}%"
+            >
+              ${score >= 50 ? '' : `${Math.floor(score)}%`}
+            </div>
+          </div>`}
```

Comment:
Once #10056 is merged this can be converted to use the `Scorebar` component. If this is merged before that PR, the change can be done there.

Reply:
It looks like this was done in 76c67db.